src/main/java/ch/njol/skript/effects/EffBan.java

```diff
@@ -73,15 +76,18 @@
 	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
 		players = exprs[0];
 		reason = exprs.length > 1 ? (Expression<String>) exprs[1] : null;
+		expires = exprs.length > 1 ? (Expression<Timespan>) exprs[2] : null;
 		ban = matchedPattern % 2 == 0;
 		ipBan = matchedPattern >= 2;
 		return true;
 	}
 	
+	@SuppressWarnings("null")
 	@Override
 	protected void execute(final Event e) {
 		final String reason = this.reason != null ? this.reason.getSingle(e) : null; // don't check for null, just ignore an invalid reason
-		final Date expires = null;
+		Timespan ts = this.expires != null ? this.expires.getSingle(e) : null;
+		final Date expires = ts != null ? new Date(System.currentTimeMillis() + ts.getMilliSeconds()) : null;

```

Comment:
is it really a good idea to permaban people if the timespan happened to be not set?

Reply:
I mean, in all fairness that is up to the Skripter to decide how they want to ban. If they don't choose a timespan then they know its going to be permanent.

Reply:
yeah but with this `ban player for {_null var}` is a permaban

Reply:
Ah good point. Do you have a suggestion for a better way around this?

Reply:
just return like other effects do? i feel like it's slightly better to not ban them rather than perma them but i'm not sure... at least it'd be consistent

Reply:
alternatively we can implement dice using armor stands and force the banned players to roll them in order to determine how long they get banned for.

i like that idea

Reply:
> alternatively we can implement dice using armor stands and force the banned players to roll them in order to determine how long they get banned for.
> 
> i like that idea

hahahahaha, OMG. 

> just return like other effects do? i feel like it's slightly better to not ban them rather than perma them but i'm not sure... at least it'd be consistent

sounds good. Ill start working on that.

Reply:
Ok I added a check for the time span equaling null. Im also throwing a small error to the console to let them know they were not successfully banned. 
Im 50/50 on the error, due to the fact that many server owners/staff don't actually look at the console and this message may be useless. 
What is your input on that? If you think I should remove it, I can remove it.

Reply:
Generally, errors written to console are not very useful. We should instead update the documentation to indicate what happens if ban length is not given. That, or permaban the players; I assume that most server owners have *some* way for players to appeal bans.

Reply:
Ok, I will definitely remove the message. 
Im more in favour of perma banning the player vs no ban at all, so I will revery back to that.

Reply:
Ok, I have applied said changes.